project/adc_driver/src/main.c

```diff
@@ -0,0 +1,51 @@
+#include <stdio.h>
+
+#include "adc.h"
+#include "interrupt.h"
+#include "log.h"
+#include "gpio.h"
+
+void test_callback(ADCChannel adc_channel, uint16_t reading, void *context) {
+  *(uint16_t*)context = reading;
```

Comment:
We generally split casting the context pointer and assigning it a value into two separate operations for clarity.